pkg/server/asset/cloud.go

```diff
@@ -0,0 +1,218 @@
+	host string,
+	authToken string,
+	publicURLPrefix string,
+	privateURLPrefix string,
+	public bool,
+) (*CloudStore, error) {
+	if appName == "" {
+		return nil, errors.New("Missing app name for cloud asset")
+	}
+
+	if host == "" {
+		return nil, errors.New("Missing host for cloud asset")
+	}
+
+	if authToken == "" {
+		return nil, errors.New("Missing auth token for cloud asset")
+	}
+
+	if public && publicURLPrefix == "" {
+		return nil, errors.New("Missing public URL prefix for cloud asset")
+	}
+
+	if !public && privateURLPrefix == "" {
+		return nil, errors.New("Missing private URL prefix for cloud asset")
+	}
+
+	urlPrefix := privateURLPrefix
+	if public {
+		urlPrefix = publicURLPrefix
+	}
+
+	store := &CloudStore{
+		appName:   appName,
+		host:      host,
+		authToken: authToken,
+		public:    public,
+		urlPrefix: urlPrefix,
+	}
+
+	log.
+		WithField("cloud-store", store).
+		Info("Created Cloud Asset Store")
+
+	// setup ticker to refresh signer token
+	store.signerTokenMutex = &sync.RWMutex{}
+	store.signerTokenRefreshTicker = time.NewTicker(
+		cloudAssetSignerTokenRefreshInterval,
+	)
+	go func(s *CloudStore) {
+		for tickerTime := range s.signerTokenRefreshTicker.C {
+			log.
+				WithField("time", tickerTime).
+				Info("Cloud Asset Signer Token Refresh Ticker Trigger")
+
+			s.refreshSignerToken()
+		}
+	}(store)
+	go store.refreshSignerToken()
+
+	return store, nil
+}
+
+func (s *CloudStore) refreshSignerToken() {
+	log.Info("Start refresh Cloud Asset Signer Token")
+
+	urlString := strings.Join(
+		[]string{s.host, "token", s.appName},
+		"/",
+	)
+	expiredAt := time.Now().
+		Add(cloudAssetSignerTokenExpiryInterval).
+		Unix()
+
+	req := goreq.Request{
+		Uri:     urlString,
+		Timeout: 10 * time.Second,
+		QueryString: struct {
+			ExpiredAt int64 `url:"expired_at"`
+		}{expiredAt},
+	}.WithHeader("Authorization", "Bearer "+s.authToken)
+
+	res, err := req.Do()
+	if err != nil {
+		log.
+			WithField("url", urlString).
+			WithField("expired-at", expiredAt).
+			WithField("error", err).
+			Error("Fail to request to refresh Cloud Asset Signer Token")
+
+		return
+	}
+
+	resBody := refreshSignerTokenResponse{}
+	err = res.Body.FromJsonTo(&resBody)
+	if err != nil {
+		log.
+			WithField("error", err).
+			WithField("response", res.Body).
+			Error("Fail to parse the response for refresh Cloud Asset Signer Token")
+
+		return
+	}
+
+	log.
+		WithField("response", resBody).
+		Info("Successfully got new Cloud Asset Signer Token")
+
+	s.signerTokenMutex.Lock()
+	s.signerToken = resBody.Value
+	s.signerExtra = resBody.Extra
+	s.signerTokenExpiredAt = resBody.ExpiredAt
+	s.signerTokenMutex.Unlock()
+}
+
+// GetFileReader returns a reader for reading files
+func (s CloudStore) GetFileReader(name string) (io.ReadCloser, error) {
+	return nil, errors.New(
+		"Directly getting files is not available for cloud-based asset store",
+	)
+}
+
+// PutFileReader return a writer for uploading files
+func (s CloudStore) PutFileReader(
```

Comment:
I am not fond of the style to have every parameter on separate line.

Are you aiming at a line-width limit?